### PR TITLE
Migration/stave

### PR DESCRIPTION
**Pull request by rvilarl**
Stave is also there :)

### CODE REVIEW

**Comment by rvilarl** [comment]
@0xfe this one is not very challenging, hopefully. :)

**Comment by rvilarl** [comment]
Obviously not :)

File: src/font.ts [hunk 1]
```diff
@@ -7,6 +7,15 @@ import { PetalumaMetrics } from './fonts/petaluma_metrics';
 import { CustomFont } from './fonts/custom_glyphs';
 import { CustomMetrics } from './fonts/custom_metrics';
 
+export interface FontStruct {
```
Comment (0xfe) [inline src/font.ts:10]: This is related to text fonts, so prob not the right place (which is music glyphs.) It's a good struct though, maybe better in `common`, also you can call it `FontInfo`.

File: src/font.ts [hunk 2]
```diff
@@ -7,6 +7,15 @@ import { PetalumaMetrics } from './fonts/petaluma_metrics';
...
+  size: number;
+  weight: string;
+  family: string;
+  style?: string;
+  font?: string;
+  face?: string;
```
Comment (0xfe) [inline src/font.ts:16]: `family`, `font`, and `face` are the same, also `point` and `size`. Merge them? or are they used differently?

Comment (rvilarl) [reply src/font.ts]: Done

File: src/font.ts (hunk 1 above)
Comment (rvilarl) [reply src/font.ts]: Done

**Comment by rvilarl** [comment]
Ready hopefully. :)

File: src/stave.ts [hunk 3]
```diff
@@ -14,9 +14,69 @@ import { Clef } from './clef';
...
+  space_above_staff_ln: number;
+  vertical_bar_width: number;
+  fill_style: string;
+  left_bar: boolean;
+  right_bar: boolean;
+  spacing_between_lines_px: number;
+  top_text_position: number;
+  num_lines: number;
+  size?: string;
+  renderer?: FactoryRendererOptions;
```
Comment (0xfe) [inline src/stave.ts:44]: Not sure why this is here.

File: src/stave.ts [hunk 4]
```diff
@@ -14,9 +14,69 @@ import { Clef } from './clef';
...
+  stave?: FactoryStaveOptions;
```
Comment (0xfe) [inline src/stave.ts:45]: This too.

File: src/stave.ts [hunk 5]
```diff
@@ -14,9 +14,69 @@ import { Clef } from './clef';
...
+  font?: FontInfo;
```
Comment (0xfe) [inline src/stave.ts:46]: I don't see this used anywhere.

File: src/stave.ts [hunk 6]
```diff
@@ -14,9 +14,69 @@ import { Clef } from './clef';
...
+  autoStem?: boolean;
+  secondaryBeamBreaks?: never[];
+  direction?: number;
+  harsh?: boolean;
+  superscript?: string;
+  position?: number;
+  style?: string;
```
Comment (0xfe) [inline src/stave.ts:53]: Many of these aren't stave options, not sure how they ended up here.

File: src/stave.ts [hunk 7]
```diff
@@ -14,9 +14,69 @@ import { Clef } from './clef';
...
+}
+
+export interface StaveTextOptions {
```
Comment (0xfe) [inline src/stave.ts:56]: This can be inlined.

File: src/types/common.d.ts [hunk 8]
```diff
@@ -18,6 +46,13 @@ export interface KeyProps {
   displaced: boolean;
 }
 
+export interface Tempo {
```
Comment (0xfe) [inline src/types/common.d.ts:49]: This should be `StaveTempoOptions` and belongs in `stavetempo.ts`

**Review by 0xfe — CHANGES_REQUESTED** [review]

File: src/stave.ts (hunk 3 above)
Comment (rvilarl) [reply src/stave.ts:44]: removed

File: src/stave.ts (hunk 5 above)
Comment (rvilarl) [reply src/stave.ts:46]: Unused elements removed.

File: src/stave.ts (hunk 6 above)
Comment (rvilarl) [reply src/stave.ts:53]: removed

File: src/stave.ts (hunk 7 above)
Comment (rvilarl) [reply src/stave.ts:56]: done

File: src/types/common.d.ts (hunk 8 above)
Comment (rvilarl) [reply src/types/common.d.ts:49]: done

File: src/stave.ts (hunk 4 above)
Comment (rvilarl) [reply src/stave.ts]: done

**Comment by rvilarl** [comment]
Hopefully now ready.

**Comment by 0xfe** [comment]
I'm guessing `package.json` here was unintentional. Can you remove?

**Comment by rvilarl** [comment]
> I'm guessing `package.json` here was unintentional. Can you remove?

Removed :)

File: src/types/common.d.ts [hunk 9]
```diff
@@ -1,3 +1,31 @@
+export interface FontInfo {
+  size: number;
+  weight: string;
+  family: string;
+}
+
+/** TODO: Move to Factory */
+export interface FactoryStaveOptions {
```
Comment (0xfe) [inline src/types/common.d.ts:8]: You can remove this.

File: src/types/common.d.ts [hunk 10]
```diff
@@ -1,3 +1,31 @@
...
+  space: number;
+}
+
+/** TODO: Move to Factory */
+export interface FactoryRendererOptions {
```
Comment (0xfe) [inline src/types/common.d.ts:13]: You can remove this too.

File: src/types/common.d.ts (hunk 9 above)
Comment (rvilarl) [reply src/types/common.d.ts]: done

File: src/types/common.d.ts (hunk 10 above)
Comment (rvilarl) [reply src/types/common.d.ts]: done

**Comment by rvilarl** [comment]
This one should be ready to merge

**Comment by rvilarl** [comment]
@0xfe please merge this one it introduces no visual change against `master`

**Review by 0xfe — APPROVED** [review]
Looks good. Merging.

**Pull request merged**